tests/gccgo/task.yaml

```diff
@@ -0,0 +1,14 @@
+summary: Check that snapd builds with gccgo
+prepare: |
+    echo Installing gccgo-6 and pretending it is the default go
+    sudo apt install -y gccgo-6
+    sudo ln -s /usr/bin/go-6 /usr/local/bin/go
+restore: |
+    rm -f /usr/local/bin/go
```

Comment:
This is actually a good practice for restore scripts, @fgimenez. The restore should be able to restore the state even if the _prepare script_ has failed and thus been partially applied. If the restore side breaks, spread will stop _completely_ because the system is now on an unknown and unreliable state that could affect further tests.